### ISSUE TITLE
Boston.py tutorial runs but, no predictions produced

### ISSUE THREAD

**Issue by dartdog** [tensorflow/tensorflow#5098]
Hard to get going when Tutorial does not work as expected..  I'm sure it is simple but not for me yet!
NOTE: Only file GitHub issues for bugs and feature requests.  All other topics will be closed.

For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### Environment info
Operating System:
Ubuntu 16.05 Cudnn 5.1 Cuda 8 Nvidia 1070

2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.

    I tensorflow/stream_executor/dso_loader.cc:116] successfully opened CUDA library libcublas.so locally
    I tensorflow/stream_executor/dso_loader.cc:116] successfully opened CUDA library libcudnn.so locally
    I tensorflow/stream_executor/dso_loader.cc:116] successfully opened CUDA library libcufft.so locally
    I tensorflow/stream_executor/dso_loader.cc:116] successfully opened CUDA library libcuda.so.1 locally
    I tensorflow/stream_executor/dso_loader.cc:116] successfully opened CUDA library libcurand.so locally
0.11.0rc0

Just running the Boston.py tutorial as is from the current repo unmodified:
It seems to train But just produces an estimator object and does not produce predictions, no errors (the warning are a bit of a concern.
this is the last train cycle and the report of the estimator object (see end).

    : GeForce GTX 1070, pci bus id: 0000:01:00.0)
    INFO:tensorflow:Restored model from /tmp/tmpq10r1l1a
    INFO:tensorflow:Eval steps [0,1) for training step 5000.
    INFO:tensorflow:Saving evaluation summary for 5000 step: loss = 21.8338
    Loss: 21.833767
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column.       Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    WARNING:tensorflow:Rank of input Tensor (1) should be the same as output_rank (2) for column. Will attempt to expand dims. It is highly recommended that you resize your input, as this behavior may change.
    Predictions: <generator object BaseEstimator._infer_model_as_iterable at 0x7f34042198e0>

**Comment by asimshankar** [comment]
@dartdog : Sorry you're having trouble getting started. Hopefully we can sort this out and make things better for future newbies.

I'm unable to reproduce the error. I tried both python 2.7 and python 3.5 and got something like:
```
Predictions: [ 33.93679047  19.3759346   21.71503448  36.20479584  15.1063385  18.72967148]
```

(You mentioned that you're using Ubuntu 16.05, but I'm guessing you meant 16.04?)
For example, here's an isolated, reproducible environment using docker and no GPU (just because the environment is easier to reproduce):
```
docker run -it  ubuntu:16.04 bash 
apt-get update
apt-get install -y python3 python3-dev python3-pandas python3-pip wget
pip3 install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.11.0rc0-cp35-cp35m-linux_x86_64.whl
mkdir /tmp/boston
cd /tmp/boston
# Download the files
wget http://download.tensorflow.org/data/boston_train.csv
wget http://download.tensorflow.org/data/boston_test.csv
wget http://download.tensorflow.org/data/boston_predict.csv
wget https://raw.githubusercontent.com/tensorflow/tensorflow/r0.11/tensorflow/examples/tutorials/input_fn/boston.py
python3 boston.py
```

Are you able to try this out? If it works then it would suggest something in your environment. If not, then we'll have more information to debug :)

**Comment by dartdog** [comment]
Hmm that ran correctly
Yes it is Ubuntu 16.04
Well as is my habit I made a typo on copying your code and so I pasted it all in
The docker image failed so I updated my base system..
So It reinstalled Python and tensorflow.. It seems the main thing is that it is now no longer using the GPU...
Dumb of me
It still seems to be using the Anaconda Python 3.5.2 (both "python' and "python3' bring the same Anaconda up in terminal) though but Tensorflow is no longer GPU enabled?
So I guess I need to reinstall it from source? Advice??
And I'm guesssing that my error may be due to the GPU enabling?

**Comment by asimshankar** [comment]
I also tried this on Ubuntu 14.04 and CUDA 7.5 (thus GPU)
(There is no release pip wheel for TensorFlow with CUDA 8 yet, hopefully it will be out soon - we intend 0.11rc1 to support CUDA 8, so I tried CUDA 7.5. And there are no nvidia/cuda docker images for CUDA 7.5 on Ubuntu 16.04, so I ended up with 14.04)

Here's what I did:
```
nvidia-docker run -it nvidia/cuda:7.5-cudnn5-devel
apt-get update
apt-get install -y python3 python3-dev python3-pandas python3-pip wget
pip3 install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.11.0rc0-cp34-cp34m-linux_x86_64.whl
mkdir /tmp/boston
cd /tmp/boston
# Download the files
wget http://download.tensorflow.org/data/boston_train.csv
wget http://download.tensorflow.org/data/boston_test.csv
wget http://download.tensorflow.org/data/boston_predict.csv
wget https://raw.githubusercontent.com/tensorflow/tensorflow/r0.11/tensorflow/examples/tutorials/input_fn/boston.py
python3 boston.py
```

The fact that this works, suggests that it isn't a GPU problem per se.
To summarize, we found that:
- CPU/Ubuntu16.04: Works
- GPU/Ubuntu14.04/CUDA7.5: Works

But we're seeing that whatever your setup (using anaconda) with Ubuntu16.04, CUDA8, Python 3.5 doesn't seem to work? It could be that specific combination, but I'd be surprised.

Could you provide details on how you setup your environment?

**Comment by dartdog** [comment]
This prior problem ticket has all my configuration details and history!
https://github.com/tensorflow/tensorflow/issues/5017
So I'll await your instructions, I'm happy to try and build gain for cuda8 ? Since I got this spiffy new board..

**Comment by asimshankar** [comment]
That setup involves copying shared libraries into the anaconda environment. I'm not well versed with anaconda enough to know what effect that may have.

Can you use docker instead? Either use the release versions (as I mentioned above for Ubuntu14.04, CUDA 7.5) or build from source using `nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04` as the base image.

I apologize that I'm basically suggesting "do something else with docker", but at this point it seems that you've run into some trouble and customizations with Anaconda that is beyond my ability to debug without being able to reproduce that exact environment.

**Comment by dartdog** [comment]
Well perhaps If I build from source in a Python 3.5 env  rather than in the root Anaconda . Things might be better isolated.. and easier to create new environments.. going forward. I'll try that out tomorrow and report back :+1:

**Comment by dartdog** [comment]
So I started form Scratch (whole new Anaconda (wiped old one completely) And re-cloned tensorFlow
Then tried to configure TF and got this error (file not found) Including more to show the rest seems ok?
The env(tf11g) is a 3.5 Anaconda Python.... It might just be a transitory error from github re protobuff bzl?

(tf11g) tom@tomServal:~/Documents/tensorflow$ ./configure
~/Documents/tensorflow ~/Documents/tensorflow
Please specify the location of python. [Default is /home/tom/anaconda3/envs/tf11g/bin/python]: 
Do you wish to build TensorFlow with Google Cloud Platform support? [y/N] y
Google Cloud Platform support will be enabled for TensorFlow
Do you wish to build TensorFlow with Hadoop File System support? [y/N] N
No Hadoop File System support will be enabled for TensorFlow
Found possible Python library paths:
  /home/tom/anaconda3/envs/tf11g/lib/python3.5/site-packages
Please input the desired Python library path to use.  Default is [/home/tom/anaconda3/envs/tf11g/lib/python3.5/site-packages]

/home/tom/anaconda3/envs/tf11g/lib/python3.5/site-packages
Do you wish to build TensorFlow with GPU support? [y/N] y
GPU support will be enabled for TensorFlow
Please specify which gcc should be used by nvcc as the host compiler. [Default is /usr/bin/gcc]: 
Please specify the Cuda SDK version you want to use, e.g. 7.0. [Leave empty to use system default]: 
Please specify the location where CUDA  toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 
Please specify the Cudnn version you want to use. [Leave empty to use system default]: 
Please specify the location where cuDNN  library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 
Please specify a list of comma-separated Cuda compute capabilities you want to build with.
You can find the compute capability of your device at: https://developer.nvidia.com/cuda-gpus.
Please note that each additional compute capability significantly increases your build time and binary size.
[Default is: "3.5,5.2"]: 
.......
INFO: Starting clean (this may take a while). Consider using --expunge_async if the clean takes more than several minutes.
.....
ERROR: error loading package 'tensorflow/core/distributed_runtime': Encountered error while reading extension file 'protobuf.bzl': no such package '@protobuf//': Error downloading from http://github.com/google/protobuf/archive/c2b3e70efd2038a54ef8973771ac58192885125e.tar.gz to /home/tom/.cache/bazel/_bazel_tom/eeeab43b7e56fc5504afeca878d5d54d/external/protobuf: Error downloading http://github.com/google/protobuf/archive/c2b3e70efd2038a54ef8973771ac58192885125e.tar.gz to /home/tom/.cache/bazel/_bazel_tom/eeeab43b7e56fc5504afeca878d5d54d/external/protobuf/c2b3e70efd2038a54ef8973771ac58192885125e.tar.gz: Failed to connect to http://github.com/google/protobuf/archive/c2b3e70efd2038a54ef8973771ac58192885125e.tar.gz : github.com.

**Comment by dartdog** [comment]
Ok once the internet cleared enough today I completely wiped my Anaconda and Tensorflow from the system and started over I created a new env just for GPU tensorFlow and reinstalled and built from scratch and once I used the solution in the other post to copy the missing library all is right with the world..  The Boston.py file runs correctly to completion and produces the scores. 

Not sure who should make sure this library below is in the right spot though..between TensorFlow and Anaconda?

    # cp /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /home/tom/anaconda3/lib/
(modify to install in the env lib directory)

Feel free to close.

**Comment by asimshankar** [comment]
I'm a little worried about having to copy over libstdc++, particularly because it means that we're changing the library that programs expecting to use the anaconda version. You might want to check in on the anaconda forums about how to use an updated libstdc++ there.

But....glad to hear things are working.

**Comment by wenzishou** [comment]
I had the same problem, then  found  if  i remove real_valued_column from **wide**,  the warning disappeared, maybe the wide part only accept ont-hot like column
hope that helpful
